### PR TITLE
Wishlist management

### LINKED ISSUE #11: Add and Remove items from wishlist for logged in user

### LINKED ISSUE #12: UI for wishlist
Add JSX mobile responsive pages

### PR DESCRIPTION
**Pull request by romabulani**
- Added UI for wishlist 
- Handled both cases when wishlist is empty and not empty
- Simultaneous changes in product listing and cart pages
- network calls for wishlist

[Deploy preview](https://deploy-preview-13--bakinlane-react.netlify.app/)

### CODE REVIEW

File: src/components/productListing/Products.jsx [hunk 1]
```diff
@@ -23,6 +23,23 @@ function Products() {
...
+  const isWishlisted = (product) =>
+    state.wishlist.find((item) => item._id === product._id);
+
+  const wishlistHandler = async (product) => {
+    if (!authToken) navigate("/login");
+    else {
+      let response;
+      if (isWishlisted(product))
+        response = await removeFromWishlist(product._id, authToken);
+      else response = await addToWishlist(authToken, product);
```
Comment (manish591) [inline src/components/productListing/Products.jsx]: Use could use ternary operator if you are not adding curly braces. But, it's up to you, if you like this approach keep it.

File: src/components/wishlist/WishlistItem.jsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  const { authToken } = useAuth();
+  const navigate = useNavigate();
+
+  const getButtonText = (item) => {
+    const filteredItem = state.cart.filter(
+      (cartItem) => item._id === cartItem._id
+    );
+    if (filteredItem.length > 0) return "Go To Cart ->";
+    else return "Add To Cart";
+  };
```
Comment (manish591) [inline src/components/wishlist/WishlistItem.jsx]: Here you should definitely use the ternary operator.

File: src/services/wishlistService.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  } catch (e) {
+    console.log("getWishlist : Error in fetching wishlist details", e);
+  }
+};
+
+const addToWishlist = async (authToken, product) => {
+  try {
+    const response = await axios.post(
+      "/api/user/wishlist",
+      { product: product },
```
Comment (manish591) [inline src/services/wishlistService.js]: You can just write product here, use object shorthand

**Review by manish591 — APPROVED** [review]
Your code is so clean and easier to read: 
- I like how you created separate functions to avoid repeatability.
- Your function naming is good. Easily understandable

**Review by ravikumar1002 — APPROVED** [review]
Good folder structure and clean code
Nice naming conventions
good  for merge

**Comment by romabulani** [comment]
> Good folder structure and clean code Nice naming conventions good for merge

Thankyou

**Comment by romabulani** [comment]
> Your code is so clean and easier to read:
> 
> * I like how you created separate functions to avoid repeatability.
> * Your function naming is good. Easily understandable

Thankyou for the suggestion, made the changes :)

**Pull request merged**